File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.js
const fs = require("fs");

let lines = fs.readFileSync(0).toString().trim().split(/\s+/);
lines.shift();

for (str of lines) {
    let [cnt, score] = [0, 0];
    for (i of str){
        if (i == 'O') {
            cnt++;
        }
        else {
            score = score + cnt * (cnt + 1) / 2;
            cnt = 0;
        }
    }
    score = score + cnt * (cnt + 1) / 2;
    console.log(score);
}    
    

File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.cc
#include <iostream>
#include <string>

using namespace std;

int n, score, cnt;
string str;

int main(void)
{
    cin >> n;
    while(n--)
    {
        cnt = 0;
        score = 0;
        cin >> str;
        for (int i = 0; i < str.length(); i++)
        {
            if (str[i] == 'O')
            {
                cnt++;
                score = score + cnt;
            }
            else
                cnt = 0;
        }
        cout << score << '\n';
    }
}